File: observable-states/contracts-java/src/main/java/com/observable/contracts/HighlyRegulatedContract.java
package com.observable.contracts;

import net.corda.core.contracts.CommandData;
import net.corda.core.contracts.Contract;
import net.corda.core.transactions.LedgerTransaction;
import org.jetbrains.annotations.NotNull;

public class HighlyRegulatedContract implements Contract {
    public static final String ID = "com.observable.contracts.HighlyRegulatedContract";

    @Override
    public void verify(@NotNull LedgerTransaction tx) throws IllegalArgumentException {
        // The contract logic is irrelevant for this sample.
    }

    public interface Commands extends CommandData {
        class Trade implements Commands {}
    }

}

File: observable-states/contracts-kotlin/src/main/kotlin/com/observable/contracts/HighlyRegulatedContract.kt
package com.observable.contracts

import net.corda.core.contracts.CommandData
import net.corda.core.contracts.Contract
import net.corda.core.transactions.LedgerTransaction

class HighlyRegulatedContract : Contract {
    companion object {
        const val ID = "com.observable.contracts.HighlyRegulatedContract"
    }

    override fun verify(tx: LedgerTransaction) {
        // The contract logic is irrelevant for this sample.
    }

    // Used to indicate the transaction's intent.
    interface Commands : CommandData {
        class Trade : Commands
    }
}